### PR TITLE
Add TypeScript types to `FilterableMultiSelect`

### LINKED ISSUE #12546: Add TypeScript types to FilterableMultiSelect / use `useCombobox` instead of `<Downshift>` component

### PR DESCRIPTION
**Pull request by guidari**
Closes #12546

#### Changelog

**New**

- Added `useCombobox` from Downshift

**Changed**

- Converted FilterableMultiSelect file from JS to TSX.

**Removed**

- Removed `<Downshift/>` component

#### Testing / Reviewing

- Make sure the functionality remains the same.
- Check if the types make sense and we didn't miss anything.
- CI checks should pass

### CODE REVIEW

**Comment by tw15egan** [comment]
@guidari looks good for the most part! I noticed that I cannot open the menu by pressing <kbd>ArrowDown</kbd> after these changes. Not sure why our tests didn't catch this, as we have this specified on `L193` in `e2e/components/Multiselect-test.avt.e2e.js`

**Comment by guidari** [comment]
> @guidari looks good for the most part! After these changes, I noticed that I could not open the menu by pressing ArrowDown. Not sure why our tests didn't catch this, as we have this specified on `L193` in `e2e/components/Multiselect-test.avt.e2e.js`

Fixed @tw15egan . The tests didn't catch this because the test was marked as `test.slow()` I just removed this and now `@avt-keyboard-nav` is running as expected.

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

File: packages/react/src/components/MultiSelect/FilterableMultiSelect.tsx [hunk 1]
```diff
@@ -0,0 +1,1011 @@
...
+  /**
+   * **Experimental**: Provide a `Slug` component to be rendered inside the `Checkbox` component
+   */
+  slug?: ReactNodeLike;
+
+  /**
+   * Provide text to be used in a `<label>` element that is tied to the
+   * combobox via ARIA attributes.
+   */
+  titleText?: ReactNode;
```
Comment (tay1orjones) [inline packages/react/src/components/MultiSelect/FilterableMultiSelect.tsx:249]: This needs to be added to the proptypes as well, https://github.com/carbon-design-system/carbon/issues/16108

File: packages/react/src/components/ListBox/next/ListBoxTrigger.js [hunk 2]
```diff
@@ -27,11 +27,11 @@ const defaultTranslateWithId = (id) => defaultTranslations[id];
  * `ListBoxTrigger` is used to orient the icon up or down depending on the
  * state of the menu for a given `ListBox`
  */
-const ListBoxTrigger = ({
-  isOpen,
-  translateWithId: t = defaultTranslateWithId,
-  ...rest
-}) => {
+// const OverflowMenu = React.forwardRef(function OverflowMenu(
```
Comment (tay1orjones) [inline packages/react/src/components/ListBox/next/ListBoxTrigger.js]:
```suggestion
```

**Review by tay1orjones — APPROVED** [review]

File: packages/react/src/components/MultiSelect/FilterableMultiSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,1018 @@
...
+    const nextIsOpen = forceIsOpen ?? !isOpen;
+    setIsOpen(nextIsOpen);
+    if (onMenuChange) {
+      onMenuChange(nextIsOpen);
+    }
+  }
+
+  // function handleOnOuterClick() {
+  //   handleOnMenuChange(false);
+  // }
```
Comment (tay1orjones) [inline packages/react/src/components/MultiSelect/FilterableMultiSelect.tsx:420]: @gui just want to double check this is no longer needed? Maybe we should add support for onChange to be logged to the actions pane in one of the storybook stories so this is testable in the deploy preview?

Comment (tay1orjones) [inline packages/react/src/components/MultiSelect/FilterableMultiSelect.tsx]: @guidari  just want to double check this is no longer needed? Maybe we should add support for onChange to be logged to the actions pane in one of the storybook stories so this is testable in the deploy preview?

**Comment by guidari** [comment]
Hey @tay1orjones we don't need that anymore because we are passing here:

```
} = useSelection({
    disabled,
    initialSelectedItems,
    onChange,
    selectedItems: selected,
  });

```
I added to the storybook

Screenshot 2024-05-08 at 15 37 19

**Comment by tay1orjones** [comment]
@guidari awesome, lets ship it! :shipit:

**Pull request merged**